presto-raptor/src/main/java/com/facebook/presto/raptor/systemtables/ShardMetadataRecordCursor.java

```diff
@@ -328,6 +330,28 @@ else if (temporalType.equals(TIMESTAMP)) {
         return builder.build();
     }
 
+    static private void buildPredicatesAndValues(Domain nameDomain, List<String> values, List<String> predicates,
+                                                 String predicatePrefix) {
+        if (nameDomain.isSingleValue()) {
+            predicates.add(predicatePrefix + " = ?");
+            values.add(getStringValue(nameDomain.getSingleValue()));
+        } else if (nameDomain.getValues() instanceof SortedRangeSet) {
```

Comment:
We shouldn't do instance checks for tuple domain classes, as that digs into the internals and violates the encapsulations. Instead, we can use the public API to check the desired properties.